packages/browser/src/__tests__/surveys.test.ts

```diff
@@ -1537,4 +1537,33 @@ describe('surveys', () => {
             expect(() => getNextSurveyStep(survey, 0, 'some_string')).toThrow('The response type must be an integer')
         })
     })
+
+    describe('is enabled', () => {
+        it('starts effectively disabled', () => {
+            expect(surveys['_isSurveysEnabled']).toBe(undefined)
+        })
+
+        it('is enabled by having results in onRemoteConfig', () => {
+            expect(surveys['_isSurveysEnabled']).toBe(undefined)
+            surveys.onRemoteConfig({
+                surveys: ['example'],
+            } as Partial<RemoteConfig> as RemoteConfig)
+            expect(surveys['_isSurveysEnabled']).toBe(true)
+        })
+
+        it('is disabled by having no results in onRemoteConfig', () => {
+            surveys.onRemoteConfig({
+                surveys: false,
+            } as Partial<RemoteConfig> as RemoteConfig)
+            expect(surveys['_isSurveysEnabled']).toBe(false)
+        })
+
+        it('can be disabled by config despite results of onRemoteConfig', () => {
+            surveys['_instance'].config.disable_surveys = true
+            surveys.onRemoteConfig({
+                surveys: ['example'],
+            } as Partial<RemoteConfig> as RemoteConfig)
+            expect(surveys['_isSurveysEnabled']).toBe(false)
```

Comment:
so, this should fail as things are now
because surveys will be enabled
because there was a response
onRemoteConfig is called regardless of other configuration so this code needs to be safe

Reply:
"needs"
given i don't know that this actually bubbles up to the user